### ISSUE TITLE
Spin out Leaflet.SpreadsheetLayer as a stand-alone Leaflet library

### ISSUE THREAD

**Issue by jywarren** [publiclab/leaflet-environmental-layers#121]
@kevinzluo made a fantastic component for this library which really deserves to be its own [Leaflet Plugin](https://leafletjs.com/plugins.html). Let's do it! 

See a recent minimal Leaflet plugin I created here for reference: https://github.com/publiclab/leaflet-multispectral

Read more here: https://github.com/publiclab/leaflet-environmental-layers/blob/master/src/util/googleSpreadsheetLayer.js

We could make it in this repository: https://github.com/publiclab/leaflet-spreadsheet

Docs could be as follows, roughly:

```

### Spreadsheet-based layers

We can source locations from a spreadsheet in a format like this:

| Title  | Latitude | Longitude   | Notes             |
|--------|----------|-------------|-------------------|
| First	 | 29.671282 | -95.17829  | The first marker  |
| Second | 29.760371 | -95.504828 | The second marker |
| Third  | 29.917755 | -95.283494 | The third marker  |

The layer is constructed like this:

\```js
L.SpreadsheetLayer = L.LayerGroup.extend({
  url: 'https://docs.google.com/spreadsheets/d/14BvU3mEqvI8moLp0vANc7jeEvb0mnmYvH4I0GkwVsiU/edit?usp=sharing', // String url of data sheet
  lat: 'latitude', // name of latitude column
  lon: 'longitude', // name of longitude column
  columns: ['title', 'notes'], // Array of column names to be used
  generatePopup: function() {
    // function used to create content of popups
  },
  // imageOptions: // optional, defaults to blank
  // sheetNum: // optional, defaults to 0 (first sheet)
});
\```

Read more here: https://github.com/publiclab/leaflet-environmental-layers/blob/master/src/util/googleSpreadsheetLayer.js
```

There are other libraries that do parts of this, but I still think this is an elegant and flexible implementation that's worth spinning out:

* https://github.com/jsoma/mapsheet
* https://github.com/veproza/moldova-mapping-with-spreadsheets
* https://github.com/mapsam/gus (good documentation on proper sharing of your spreadsheet)
* https://github.com/geraldo/driveMap
* https://github.com/banmedo/LeafletDigitizer

**Comment by jywarren** [comment]
Note that I'm having trouble instantiating a new layer... not sure why it's not working. @kevinzluo why does this have to be a LayerGroup?

**Comment by kevinzluo** [comment]
Hello,
I'm so sorry, but I really haven't been active in the community lately. I'm
caught up in school and really can't take a look right now, but I will as
soon as a have the chance. There is a break coming up in mid March so I
could try then? I remember when I was creating the layer I had some issues
as well, but there was some way to do it.

My apologies!

>

**Comment by gauravano** [comment]
No issue @kevinzluo, we understand your situation and you are most welcome to contribute anytime, as per your convenience. Thanks!

**Comment by jywarren** [comment]
yes, sorry @kevinzluo i didn't mean to imply you needed to do this either!
I was just looking at projects people might be able to do to generally
improve the codebase and wanted to keep you in the loop! Good luck with
school and thanks for all your help!
